File: SDE_sheet/Arrays/21_longest_consecutive_sequence/solution.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        num_set = set(nums)
        max_length = 0
        
        for num in num_set:
            if num - 1 not in num_set:
                curr = 1
                while num + 1 in num_set:
                    num += 1
                    curr += 1
                max_length = max(max_length, curr)
        
        return max_length

File: SDE_sheet/Arrays/21_longest_consecutive_sequence/solution.rs
use std::collections::HashSet;

impl Solution {
    pub fn longest_consecutive(nums: Vec<i32>) -> i32 {
        let num_set: HashSet<i32> = nums.into_iter().collect();
        let mut max_length = 0;
        
        for &num in num_set.iter() {
            if !num_set.contains(&(num - 1)) {
                let mut curr = 1;
                let mut next_num = num + 1;
                while num_set.contains(&next_num) {
                    curr += 1;
                    next_num += 1;
                }
                max_length = max_length.max(curr);
            }
        }
        
        max_length
    }
}
